src/components/header/PageHeader.jsx

```diff
@@ -101,22 +103,29 @@ class PageHeader extends Component {
     }));
   };
 
-  renderSelectAllApprovebtn = () => (
-    <div className='pageHeader__selectAction'>
-      <input
-        type='checkbox'
-        name='checkbox'
-        className='pageHeader__selectApprove__checkbox'
-        onChange={this.props.handleSelectAllClick}
-      /> Select all
-      <Button
-        name='approveAll'
-        value='Approve Selected'
-        className='pageHeader__selectAction__button pageHeader__selectApprove__button '
-        onClick={this.props.handleApproveAllClick}
-      />
-    </div>
-  )
+  renderSelectAllApprovebtn = () => {
+    const { disabled } = this.props;
+    return (
+      <div className='pageHeader__selectAction'>
+        <input
+          type='checkbox'
+          name='checkbox'
+          className='pageHeader__selectApprove__checkbox'
+          onChange={this.props.handleSelectAllClick}
+        /> Select all
+        <Button
+          name='approveAll'
+          value='Approve Selected'
+          className={
+            disabled ?
+              'pageHeader__disable__button' : 'pageHeader__selectAction__button pageHeader__selectApprove__button '
+          }
+          onClick={this.props.handleApproveAllClick}
```

Comment:
You could also destructure the function here since you are also getting it from props.

Reply:
@kimani-ndegwa Fixed. that was an oversight

Reply:
Np. 👍